src/rust/ide/utils/src/vec.rs

```diff
@@ -16,6 +16,17 @@ pub trait VecExt<T> : AsMut<Vec<T>> {
     fn pop_front(&mut self) -> Option<T> {
         self.try_remove(0)
     }
+
+    /// Removes the last `n` elements from the vector.
+    ///
+    /// Returns Some if elements were succesfully removed.
+    /// Returns None and leaves self intact, if there were not enough elements.
+    fn remove_last_n(&mut self, n:usize) -> Option<()> {
+        let vec      = self.as_mut();
+        let new_size = vec.len().checked_sub(n)?;
+        vec.truncate(new_size);
+        Some(())
+    }
```

Comment:
I find returning option for unit kinda, hmm... strange. Maybe return bool?
```suggestion
    /// Removes the last `n` elements from the vector.
    ///
    /// Returns Some if elements were succesfully removed.
    /// Returns None and leaves self intact, if there were not enough elements.
    fn remove_last_n(&mut self, n:usize) -> bool {
        {
            let vec      = self.as_mut();
            let new_size = vec.len().checked_sub(n)?;
            vec.truncate(new_size);
            Some(())
        }.is_some()
    }
```
Not sure if it actually compiles.